### PR TITLE
update VF flag API

### PR DESCRIPTION
**Pull request by zshi-redhat**
Add support for link state, vlan qos
Update spoofChk & trust

### CODE REVIEW

File: pkg/apis/sriovnetwork/v1/sriovnetworknodepolicy_types.go [hunk 1]
```diff
@@ -41,7 +41,6 @@ type SriovNetworkNicSelector struct {
 	// +kubebuilder:validation:Enum={"8086","15b3"}
 	// The vendor hex code of SR-IoV device. Allowed value "8086", "15b3".
 	Vendor      string   `json:"vendor,omitempty"`
-	// +kubebuilder:validation:Enum=1583,158b,10fb,1015,1017
```
Comment (pliurh) [inline pkg/apis/sriovnetwork/v1/sriovnetworknodepolicy_types.go]: We can keep this line by changing the formate to `Enum={"1583","158b","10fb","1015","1017"}`

File: pkg/apis/sriovnetwork/v1/sriovnetwork_types.go [hunk 2]
```diff
@@ -19,11 +19,17 @@ type SriovNetworkSpec struct {
...
-	// VF spoof check
-	SpoofChk *bool `json:"spoofChk,omitempty"`
-	// VF trust mode
-	Trust    *bool `json:"trust,omitempty"`
+	Vlan		int   `json:"vlan,omitempty"`
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:validation:Maximum=7
+	// VLAN QoS ID to assign for the VF. Defaults to 0.
+	VlanQoS		int   `json:"vlanQoS,omitempty"`
+	// VF spoof check, (on|off)
```
Comment (pliurh) [inline pkg/apis/sriovnetwork/v1/sriovnetwork_types.go]: Please add `+kubebuilder:validation:Enum=...` for CRD validation for all the new feilds.

Comment (zshi-redhat) [reply pkg/apis/sriovnetwork/v1/sriovnetwork_types.go]: fixed

File: pkg/apis/sriovnetwork/v1/sriovnetworknodepolicy_types.go (hunk 1 above)
Comment (zshi-redhat) [reply pkg/apis/sriovnetwork/v1/sriovnetworknodepolicy_types.go]: fixed

**Pull request merged**